app/models/setting.rb

```diff
@@ -106,12 +106,8 @@ def default=(v)
   def parse_string_value(val)
     case settings_type
     when "boolean"
-      val = val.downcase
-      if val == "true"
-        self.value = true
-      elsif val == "false"
-        self.value = false
-      else
+      self.value = Foreman::Cast.to_bool(val)
+      if self.value == self.default
```

Comment:
I see your point, but should the value have been legitimately changed to the same value as the default, it's going to raise an error to the user.

Instead of using `#value=` and `#value`, best to store the to_bool result in a temporary value, check whether it's nil for the error message and then call `#value=` if it's good.